### DISCUSSION TITLE
how to feed stdin to external process and redirect stdout ?

### DISCUSSION THREAD

**Discussion by tissatussa** [can-lehmann/owlkettle#134] [Q&A]
i'm new to Owlkettle but i managed to compile most of your examples into a GUI.
i also have some experience programming Nim, but can't succeed redirecting / catching stdout.
i'm trying to interact with a chess engine binary, but the question is general : how to fill a text widget with such data ?

i found forum entries like : 

**Redirect stdout for certain execution**
https://forum.nim-lang.org/t/6909

..but i can't make those code parts work - can anyone give some basic code, or point me to a working example ?

**Comment by tissatussa** [comment]
Btw. maybe this better be a Nim question in general ? yet i have no clue about stdin/stdout regarding your widgets.

**Comment by can-lehmann** [comment]
It might be worth also asking this on the Nim Forum or the Nim Discord, as it is somewhat at the intersection of a owlkettle / Nim question. The owlkettle specific aspect of this question is probably how to asynchronously read from stdout (while the gui remains interactive). You can [achieve with this using threading](https://github.com/can-lehmann/owlkettle/blob/main/examples/misc/threading.nim).

**Comment by tissatussa** [comment]
> You can achieve with this using threading..

i looked into `threading.nim` but i see no executable being fired ?  What does the thread do here ? Is it empty ?

**Comment by tissatussa** [comment]
> asynchronously read from stdout..

i'm aware these techniques exist but i'm a simple programmer, i need examples .. then i can change things and learn.  Such (a)sync scripts are hard to understand, to me at least ..

**Comment by tissatussa** [comment]
i created a working example now : https://github.com/StefanSalewski/salewski-chess/issues/3#issuecomment-1821710907
